tasks/localizationTasks.ts

```diff
@@ -0,0 +1,174 @@
+
+import * as gulp from 'gulp';
+import * as process from 'node:process';
+import * as minimist from 'minimist';
+import { spawnSync } from 'node:child_process';
+import * as fs from 'fs';
+import * as path from 'path';
+import * as util from 'node:util';
+import { EOL } from 'node:os';
+import { Octokit } from '@octokit/rest';
+
+type Options = {
+    userName: string;
+    email: string;
+    owner: string;
+    commitSha: string;
+    targetRemoteRepo: string;
+    baseBranch: string;
+};
+
+const localizationLanguages = ['cs', 'de', 'es', 'fr', 'it', 'ja', 'ko', 'pl', 'pt-br', 'ru', 'tr', 'zh-cn', 'zh-tw'];
+const locFiles = ['bundle.l10n.%s.json', 'package.nls.%s.json'];
+
+function getGeneratedLocalizationChanges(diffFilesAndDirectories: string[]): string[] {
+    if (diffFilesAndDirectories.length == 0) {
+        return [];
+    }
+
+    const allPossibleLocalizationFileNames = getAllPossibleLocalizationFileNames();
+    const changedLocFilesOrDirectory = [];
+
+    for (const diffFileOrDirectory of diffFilesAndDirectories) {
+        const stat = fs.statSync(diffFileOrDirectory);
+        if (
+            stat.isFile() &&
+            allPossibleLocalizationFileNames.some((locFileName) => path.basename(diffFileOrDirectory) === locFileName)
+        ) {
+            console.log(`${diffFileOrDirectory} is changed.`);
+            changedLocFilesOrDirectory.push(diffFileOrDirectory);
+        }
+
+        if (stat.isDirectory() && diffFileOrDirectory !== 'l10n') {
+            console.log('l10n is changed.');
+            changedLocFilesOrDirectory.push('l10n');
+        }
+    }
+
+    return changedLocFilesOrDirectory;
+}
+
+function getAllPossibleLocalizationFileNames(): string[] {
+    const files = [];
+    for (const lang of localizationLanguages) {
+        for (const file of locFiles) {
+            files.push(util.format(file, lang));
+        }
+    }
+    // English
+    files.push('bundle.l10n.json');
+    return files;
+}
+
+async function git_diff(args: string[]): Promise<string[]> {
+    const result = await git(['diff'].concat(args));
+    // Line ending from the stdout of git is '\n' even on Windows.
+    return result
+        .replaceAll('\n', EOL)
+        .split(EOL)
+        .map((fileName, _) => fileName.trim())
+        .filter((fileName) => fileName.length !== 0);
+}
+
+async function git(args: string[], printCommand = true): Promise<string> {
+    if (printCommand) {
+        console.log(`git ${args.join(' ')}`);
+    }
+
+    const git = spawnSync('git', args);
+    if (git.status != 0) {
+        const err = git.stderr.toString();
+        if (printCommand) {
+            console.log(`Failed to execute git ${args.join(' ')}.`);
+        }
+        throw err;
+    }
+
+    const stdout = git.stdout.toString();
+    if (printCommand) {
+        console.log(stdout);
+    }
+    return stdout;
+}
+
+gulp.task('publish localization content', async () => {
```

Comment:
This task is basically performing
`git add -A` (Stage all the changed files)
`git diff HEAD` (Find all the changed file, find 'bundle.l10n.<lang>.json', 'package.nls.<lang>.json')
`git reset` (Clean all staged files)
`git add bundle.l10n.<lang>.json' package.nls.<lang>.json `

Use bot account to create a PR in GitHub

Reply:
I'm probably missing something - why do we need the first 3 steps?  Is it not possible to always just do a `git add` on all of the possible localization files?  I think git should ignore ones trhat aren't changed

Reply:
No, the loc task would generate a lot of other stuff, see here https://dev.azure.com/dnceng/internal/_build/results?buildId=2234885&view=artifacts&pathAsName=false&type=publishedArtifacts

Reply:
What I mean is we know what files we want to keep - we could always run a git add on all the files we know we want, and don't add everything else (its left as unstaged).

Basically kind of just hardcode the files we know we want
`git add xyz-cs.json`
`git add xyz-de.json`
etc.

then commit and push

Then we dont need to diff / reset or anything

Reply:
Done :)